### PR TITLE
feat: enable rule saving

### PR DESCRIPTION
**Pull request by Hellebore**
## Checklist

- [ ] If `package.json` or `yarn.lock` have changed, then test the VSIX built by `yarn run vsce package` works from a direct install

### CODE REVIEW

File: src/rule-search.ts [hunk 1]
```diff
@@ -88,7 +111,7 @@ export class RuleInputProvider implements vscode.WebviewViewProvider {
 					return languageId;
 				}
 			}
-			return languageId;
+			return "python";
```
Comment (Hellebore) [inline src/rule-search.ts:114]: default to python to stop e.g. yaml being set when opening from a config file

File: package.json [hunk 2]
```diff
@@ -207,6 +212,11 @@
 		],
 		"menus": {
 			"view/title": [
+				{
+					"command": "sourcery.scan.selectLanguage",
+					"when": "view == sourcery.rules",
+					"group": "navigation"
+				},
 				{
```
Comment (Hellebore) [inline package.json:220]: new command to choose the language in the rule scan

File: package.json [hunk 3]
```diff
@@ -234,7 +244,7 @@
 			],
 			"sourcery.editor": [
 				{
-					"command": "sourcery.rule.create",
+					"command": "sourcery.scan.open",
 					"title": "Create Sourcery rule with this pattern",
 					"when": "editorLangId==python && editorHasSelection"
```
Comment (Hellebore) [inline package.json:249]: changed the right-click option when selecting code in the editor to open the scan and copy the code into the pattern, rather than creating a rule directly.

File: src/extension.ts [hunk 4]
```diff
@@ -184,7 +197,12 @@ function registerCommands(context: ExtensionContext, riProvider: RuleInputProvid
         vscode.commands.executeCommand('setContext',
             'sourceryRulesActive',
             true);
-        vscode.commands.executeCommand("sourcery.rules.focus")
+
+        vscode.commands.executeCommand("sourcery.rules.focus").then( () => {
+            const input = getValidInput();
+            riProvider.setPattern(input);
+            }
```
Comment (Hellebore) [inline src/extension.ts:204]: if we're focusing on the rules view, copy in any selected text to the pattern

File: src/rule-search.ts [hunk 5]
```diff
@@ -18,10 +18,19 @@ export class RuleInputProvider implements vscode.WebviewViewProvider {
 		this._extensionUri = _context.extensionUri;
 	}
 
-	public async toggle() {
+	public toggle() {
```
Comment (ruancomelli) [inline src/rule-search.ts:21]: Why this change?

**Review by ruancomelli — APPROVED** [review]
Looks good to a non-expert like me :smile: 
The only major request is to use some auto-formatting tool as you suggested yesterday, but, of course, that's out of the scope of this PR.

File: src/extension.ts [hunk 6]
```diff
@@ -157,6 +157,19 @@ function registerCommands(context: ExtensionContext, riProvider: RuleInputProvid
         riProvider.toggle();
     }));
 
+    context.subscriptions.push(commands.registerCommand('sourcery.scan.selectLanguage', () => {
+        const items = ['python', 'javascript'];
```
[the remark this answers is not in the document]
Comment (Hellebore) [reply src/extension.ts:161]: can add that in later yes - and agree would be nicer to load from the LSP, but not sure how to do that.

File: src/rule-search.ts (hunk 5 above)
Comment (Hellebore) [reply src/rule-search.ts:21]: mostly realised it didn't need to be async
